dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/DatabaseConnection.kt

```diff
@@ -91,4 +92,46 @@ object DatabaseConnection {
             connectionProps,
         )
     }
```

Comment:
Wieso ist `conn` eigentlich überall nullable?
Meine Idee wäre, das non-nullable in den Operations-Funktionen zu machen.
Die Operation soll und kann nur ausgeführt werden, wenn eine connection aufgebaut werden konnte.
Wenn die connection nicht aufgebaut werden konnte, sollten wir das iwie handlen (retries?, exception?)

Reply:
Aber die gesamte Operation wird doch schon vom retryWrapper umgeben. Ich würde da jetzt kein Retry im Retry machen.

Reply:
Und es war einfacher einen nullable Return zu geben, als mit einem try/catch Block zu arbeiten.

Reply:
Also dein Vorgehen ist ja, dass du "getConnection" machst, und dann die connection in deine Datenbank-Funktionen reingibst (z.B. insertByteArrayIntoSqlDatabase).
Deine Datenbank-Funktionen akzeptieren aber alle "null" als connection. Das macht auf den ersten Blick keinen Sinn.

Mein Vorschlag ist, dass "getConnection" entweder eine connection zurückgibt, oder nicht. Aber es sollte kein "null" zurückggeben bzw. den null-Fall selbst handlen, anstatt die Datenbank-Funktionen diesen FAll handöen zu lassen.

Können auch gerne nochmal darüber sprechen.